caas/containers.go

```diff
@@ -63,33 +63,33 @@ type PodSpec struct {
 	CustomResourceDefinitions []CustomResourceDefinition `yaml:"crd,omitempty"`
 }
 
-// CrdTemplateValidation defines the custom resource definition validation schema.
-type CrdTemplateValidation struct {
+// CustomResourceDefinitionValidation defines the custom resource definition validation schema.
+type CustomResourceDefinitionValidation struct {
 	Properties map[string]apiextensionsv1beta1.JSONSchemaProps `yaml:"properties"`
 }
 
 // CustomResourceDefinition defines the custom resource definition template and content format in podspec.
 type CustomResourceDefinition struct {
-	Kind       string                `yaml:"kind"`
-	Group      string                `yaml:"group"`
-	Version    string                `yaml:"version"`
-	Scope      string                `yaml:"scope"`
-	Validation CrdTemplateValidation `yaml:"validation,omitempty"`
+	Kind       string                             `yaml:"kind"`
+	Group      string                             `yaml:"group"`
+	Version    string                             `yaml:"version"`
+	Scope      string                             `yaml:"scope"`
+	Validation CustomResourceDefinitionValidation `yaml:"validation,omitempty"`
 }
 
 // Validate returns an error if the crd is not valid.
 func (crd *CustomResourceDefinition) Validate() error {
 	if crd.Kind == "" {
-		return errors.New("kind is missing")
+		return errors.NotValidf("kind is missing")
```

Comment:
The Error() string from NotValid errors comes out like this "kind is missing is not valid"
So you just need error.NotValidf("missing Kind")